stream.go

```diff
@@ -67,7 +68,16 @@ func (sr *streamRecorder) Write(data []byte) (int, error) {
 		return sr.ResponseWriterWrapper.Write(data)
 	}
 
-	return sr.transaction.ResponseBodyWriter().Write(data)
+	interruption, n, err := sr.transaction.WriteResponseBody(data)
+	if err != nil {
+		return 0, fmt.Errorf("failed to append response body: %s", err.Error())
+	}
+	if interruption != nil {
+		// Interruption raised while appending response body (e.g. limit reached)
+		return 0, nil
```

Comment:
I had to fix also the breaking change about writing the response body. Not it returns also an interruption. Is it okay to just handle the interruption like this? Please, double check, first time contributing to coraza-caddy